tools/ProductConstructionService.ReproTool/Operations/Operation.cs

```diff
@@ -83,11 +95,44 @@ protected async Task UpdateVmrSourceFiles(string branch, string productRepoUri,
     {
         // Fetch source mappings and source manifest files and replace the mapping for the repo we're testing on
         logger.LogInformation("Updating source mappings and source manifest files in VMR fork to replace original product repo mapping with fork mapping");
-        await UpdateRemoteVmrForkFileAsync(branch, productRepoUri, productRepoForkUri, SourceMappingsPath);
-        await UpdateRemoteVmrForkFileAsync(branch, productRepoUri, productRepoForkUri, SourceManifestPath);
+
+        await UpdateRemoteVmrForkFileAsync(branch, SourceMappingsPath, content =>
+        {
+            var settings = JsonSerializer.Deserialize<SourceMappingFile>(content, JsonSerializerOptions)
+                ?? throw new Exception($"Failed to deserialize {VmrInfo.SourceMappingsFileName}");
+
+            foreach (var mapping in settings.Mappings)
```

Comment:
this could also be something like `settings.Mappings.First(...).DefaultRemote = productRepoForkUri`?